### PR TITLE
EDU-1216 Prevent own admin role removal

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-1216

Screen Shot 2023-03-09 at 17 27 55

### CODE REVIEW

**Comment by adrianaluchian** [comment]
_Adriana updated this Pull Request_

File: src/domain/schemas/user-schemas.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export const userDBSchema = joi.object({
   _id: idOrKeySchema.required(),
   passwordHash: joi.string().allow(null).required(),
   email: joi.string().case('lower').allow(null).required(),
-  roles: joi.array().required().items(joi.string()),
+  role: joi.string().required(),
```
Comment (ahelmberger) [inline src/domain/schemas/user-schemas.js:108]: We also could restrict here to: `joi.string().valid(...Object.values(ROLE)).required()`

**Review by ahelmberger — APPROVED** [review]

File: src/domain/schemas/user-schemas.js (hunk 1 above)
Comment (adrianaluchian) [reply src/domain/schemas/user-schemas.js]: Indeed, added!
Apropo to the validity of the previous `roles` array, there were a few of empty arrays in the Prod DB 😬

**Comment by ahelmberger** [comment]
Another nice simplification :+1:

**Pull request merged**